lib/cipherstash/index/filter_match.rb

```diff
@@ -0,0 +1,52 @@
+require_relative "./bloom_filter"
+
+module CipherStash
+    class Index
+      # Implementation for the 'filter-match' index type
+      #
+      # @private
+      class FilterMatch < Index
+        INDEX_OPS = {
+          "match" => -> (idx, s) do
+            filter = BloomFilter.new(
+              [idx.meta_settings["$filterKey"]].pack("H*"),
+              idx.mapping_settings
+            )
+
+            bits = idx.text_processor.perform(s)
+              .reduce(filter) { |filter, term| filter.add(term) }
+              .bits
+              .to_a
+
+            { indexId: idx.binid, filter: { bits: bits } }
+          end,
+        }
+
+        def orderable?
+          false
+        end
+
+        def analyze(uuid, record)
+          blid = blob_from_uuid(uuid)
+
+          field_names = @settings["mapping"]["fields"]
+          raw_terms = field_names.map { |n| nested_lookup(record, n) }.compact
+
+          if raw_terms == []
+            nil
+          else
+            filter = BloomFilter.new([meta_settings["$prfKey"]].pack("H*"))
+            bits = raw_terms
+              .map { |s| text_processor.perform(s) }
+              .flatten
+              .uniq
+              .reduce(filter) { |filter, term| filter.add(term) }
+              .bits
+              .to_a
```

Comment:
The code varies slightly between the different match index classes, so it's a bit tough to abstract.

Rather that trying to pull it into a single method, what do you think about doing something like:
* Putting `[meta_settings["$filterKey"]].pack("H*")` into something like a `hex_key(key_name)` method on the `Index` class.
* Putting `.reduce(filter) { |filter, term| filter.add(term) }` into a `BloomFilter` method that adds multiple terms to the filter instead of a single term
* Adding a `to_a` method directly to `BloomFilter` so that the filter match classes don't have to grab the bits and then cast the set to a list

Reply:
Love it.

Reply:
Done: https://github.com/cipherstash/ruby-client/pull/13/commits/a2a8adf6073082c942f3e1d6f63b24a04850faac